### PR TITLE
Enable sending messages to an unlimited number of recipients

### PR DESCRIPTION
**Pull request by Ahmed-Naguib93**
refs: [MBL-17936](https://instructure.atlassian.net/browse/MBL-17936)
affects: Student , Teacher
release note: Enabled sending messages to an unlimited number of recipients

## Test Plane
- Open a new message
- Send a message to over 100 recipients.
- Observe that the "Send Individually" toggle becomes disabled.

#### Note 
- If you send a reply message to over 100 recipients, you will receive an error. This is not an issue on our side but a backend issue.

## Screenshots

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td></td>
<td></td>
</tr>

https://github.com/user-attachments/assets/8be1c566-abb7-4f2b-98c0-450c1595503b

## Checklist
- [x] A11y checked
- [ ] Tested on phone
- [x] Tested on tablet
- [x] Tested in dark mode
- [x] Tested in light mode

[MBL-17936]: https://instructure.atlassian.net/browse/MBL-17936?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Teacher Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>



#### Release Note: 
Enabled to send message over 100 recipients

#### Affected Apps: Teacher, Student

[MBL-17936](https://instructure.atlassian.net/browse/MBL-17936)

<details><summary>:x: Swift lint</summary>


    ❌ /Users/vagrant/git/Core/CoreTests/Conversations/ComposeViewControllerTests.swift:119:1: Trailing Whitespace Violation: Lines should not have trailing whitespace (trailing_whitespace)
    
    
</details>
<details><summary>:x: XCTest failed: CoreTests/CourseSyncSyllabusInteractorLiveTests/testSuccessfulFetch</summary>


    XCTAssertEqual failed: ("0") is not equal to ("2")
    XCTAssertEqual failed: ("nil") is not equal to ("Optional("1")")
    XCTAssertEqual failed: ("nil") is not equal to ("Optional("2")")
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3f7f3918e0fc8561322a4a5c9ab85a1dd452644a
</p>

File: Core/Core/Inbox/ComposeMessage/ViewModel/ComposeMessageViewModel.swift [hunk 1]
```diff
@@ -408,15 +429,17 @@ final class ComposeMessageViewModel: ObservableObject {
         if subject.isEmpty {
             subject = title
         }
-
+        let isExceedLimitRecipients = recipientIDs.count > maxRecipientCount
```
Comment (rh12) [inline Core/Core/Inbox/ComposeMessage/ViewModel/ComposeMessageViewModel.swift:432]:
```suggestion
        let isExceedsRecipientsLimit = recipientIDs.count > maxRecipientCount
```

File: Core/CoreTests/Inbox/ComposeMessage/ComposeMessageViewModelTests.swift [hunk 2]
```diff
@@ -641,6 +641,15 @@ class ComposeMessageViewModelTests: CoreTestCase {
...
         XCTAssertTrue(testee.searchedRecipients.isEmpty)
     }
+
+    func test_selectedRecipients_disableToggle() {
+        // Given
+        let recipient = ReceiptStub.getRecipientExceedMaxLimit()
+        // When
+        testee.selectedRecipients.send(recipient)
+        // Then
+        XCTAssertTrue(testee.isSendIndividualToggleDisabled)
```
Comment (rh12) [inline Core/CoreTests/Inbox/ComposeMessage/ComposeMessageViewModelTests.swift:651]: 1. I think we should also test whether `sendIndividual` is set to true in this case.
2. And maybe that it's set back to the previous value after limit is no longer exceeded.

File: Core/Core/Inbox/ComposeMessage/View/ComposeMessageView.swift [hunk 3]
```diff
@@ -307,6 +307,7 @@ public struct ComposeMessageView: View, ScreenViewTrackable {
             .font(.regular16, lineHeight: .condensed)
             .foregroundColor(.textDark)
             .padding(.vertical, 12)
+            .frame(maxHeight: .infinity, alignment: .center)
```
Comment (rh12) [inline Core/Core/Inbox/ComposeMessage/View/ComposeMessageView.swift:310]: I've asked Marci about this change, will update with his answer.
Regardless, please highlight these kind of unrelated changes in the PR description, especially if those are user facing.

**Review by rh12 — COMMENTED** [review]
QA + 1

Thanks for the changes!
Will approve once the UX change below is confirmed.

File: Core/Core/Inbox/ComposeMessage/View/ComposeMessageView.swift (hunk 3 above)
Comment (Ahmed-Naguib93) [reply Core/Core/Inbox/ComposeMessage/View/ComposeMessageView.swift:310]: yes i faced an issues when you set much recipients in the list the 'TO' button in the top and '+' in the middle, so i fixed it 
@rh12

Comment (rh12) [reply Core/Core/Inbox/ComposeMessage/View/ComposeMessageView.swift:310]: Marci confirmed it, that the "To" text should stay top aligned. Actually the "+" should also stay top aligned.
(This was probably not spotted in the previous PR)

Please revert the "To" change, and could you please fix the "+"? If so, please add that to the PR description.

Comment (Ahmed-Naguib93) [reply Core/Core/Inbox/ComposeMessage/View/ComposeMessageView.swift:310]: Thanks will do

**Review by rh12 — APPROVED** [review]
QA + 1

File: Core/Core/Inbox/ComposeMessage/ViewModel/ComposeMessageViewModel.swift [hunk 4]
```diff
@@ -103,6 +104,8 @@ final class ComposeMessageViewModel: ObservableObject {
     private var hiddenMessage: String = ""
     private var autoTeacherSelect: Bool = false
     private var teacherOnly: Bool = false
+    private var sendIndividualToggleTemp: Bool = false
```
Comment (vargaat) [inline Core/Core/Inbox/ComposeMessage/ViewModel/ComposeMessageViewModel.swift:107]: Can we have a more telling name here? Like `sendIndividualToggleLastValue`? This variable is definitely not temporary.

**Review by vargaat — APPROVED** [review]
QA+1

**Review by rh12 — APPROVED** [review]
